### ISSUE TITLE
Warn when subclassing NSCell without implementing Copy (NSZone) and an IntPtr constructor

### ISSUE THREAD

**Issue by rolfbjarne** [xamarin/xamarin-macios#3886]
NSCell is special (in a bad way): https://github.com/xamarin/maccore/commit/d762a2202ef87fa8407da5b92444dd4e144485ba

This means that any custom NSCell subclasses should provide an IntPtr constructor, and probably override Copy (NSZone) as well.

We should add a warning to mmp to tell users about this.

This is only an issue in Xamarin.Mac (Apple wisely never added the evil `NSCopyObject` method in iOS, so this problem will never show up there).

**Comment by spouliot** [comment]
I'm not sure if `mmp` is the right place for such check. But we can 

* add `[Advice]` on the API for the IDE to show at development time; and

* update the API [documentation](https://developer.xamarin.com/api/type/MonoMac.AppKit.NSCell/) to mention the requirement

**Comment by rolfbjarne** [comment]
> I'm not sure if mmp is the right place for such check.

Xamarin.Analysis maybe?

**Comment by spouliot** [comment]
no, but a roslyn analyzer could. However the complexity is much higher (than adding `[Advice]`) and it needs a new release (when updating docs does not) :)
